Backend/starter_project_dotnetforce7/BlogApp2/BlogApp.Application/Features/Blog/CQRS/Handlers/CreateBlogCommandHandler.cs

```diff
@@ -0,0 +1,54 @@
+using AutoMapper;
+using BlogApp.Application.Contracts.Persistence;
+using BlogApp.Application.Features.Blogs.CQRS.Commands;
+using BlogApp.Application.Features.Blogs.DTOs.Validators;
+using BlogApp.Application.Responses;
+using MediatR;
+using BlogApp.Domain;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace BlogApp.Application.Features.Blogs.CQRS.Handlers
+{
+    public class CreateBlogCommandHandler : IRequestHandler<CreateBlogCommand, BaseCommandResponse>
+    {
+        private readonly IUnitOfWork _unitOfWork;
+        private readonly IMapper _mapper;
+
+        public CreateBlogCommandHandler(IUnitOfWork unitOfWork, IMapper mapper)
+        {
+            _unitOfWork = unitOfWork;
+            _mapper = mapper;
+        }
+
+        public async Task<BaseCommandResponse> Handle(CreateBlogCommand request, CancellationToken cancellationToken)
+        {
+            var response = new BaseCommandResponse();
+            var validator = new CreateBlogDtoValidator();
+            var validationResult = await validator.ValidateAsync(request.BlogDto);
+
+            if (validationResult.IsValid == false)
+            {
+                response.Success = false;
+                response.Message = "Creation Failed";
+                response.Errors = validationResult.Errors.Select(q => q.ErrorMessage).ToList();
+            }
+            else
+            {
+                var blog = _mapper.Map<Blog>(request.BlogDto);
+
+                blog = await _unitOfWork.BlogRepository.Add(blog);
+                await _unitOfWork.Save();
+
+                response.Success = true;
+                response.Message = "Creation Successful";
+                response.Id = blog.Id;
```

Comment:
You should check if the saving process was successful or not. check the following snippet taken from your groups main branch:

 if (await _unitOfWork.Save() > 0)
                {
                    response.Success = true;
                    response.Message = "Creation Successful";
                    response.Value = rate.Id;
                }
                else
                {
                    response.Success = false;
                    response.Message = "Creation Failed";
                }

The implementation of unit of work in persistence layer returns integer(representing number changes made to the DB)...we can use that to know wether the process of creating a blog was successful or not.